app/services/openai_provider.py

```
@@ -0,0 +1,111 @@
+        """Validate OpenAI API key."""
+        try:
+            client = AsyncOpenAI(api_key=api_key)
+            await client.chat.completions.create(
+                model="gpt-4o-mini",
+                messages=[{"role": "user", "content": "test"}],
+                max_tokens=1
+            )
+            return True
+        except openai.AuthenticationError:
+            return False
+        except Exception:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)